### PR TITLE
switched to coverage.py

### PR DESCRIPTION
**Pull request by FrederikRothe**
## Describe your changes
Implemented the testing library coverage.py as this helps testing the code coverage and enables us to set a minimum coverage for pull requests as a GitHub action.
## Checklist before requesting a review

- [✅ ] I have performed a self-review of my code.
- [✅  ] If it is a core feature, I have added thorough tests.
- [✅  ] Have I requested and notified the boys for review of the pull-request?

### CODE REVIEW

**Comment by FrederikRothe** [comment]
Test coverage threshold can be changed by adjusting value in line 24.
image

**Review by toffernator — COMMENTED** [review]
Can we just fail under 0 for now? Don't want this to get in the way while we stabilize the program's interfaces and such

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+coverage==7.1.0
```
Comment (toffernator) [inline requirements.txt:1]: Use poetry for dependencies.

**Review by toffernator — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (FrederikRothe) [reply requirements.txt]: deleted req.txt and added coverage to poetry

File: pyproject.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.11"
+coverage = "7.1.0"
```
Comment (toffernator) [inline pyproject.toml:10]: I would make it a dev dependency

**Pull request closed without merging**